src/rust/ide/src/controller/module.rs

```diff
@@ -129,6 +127,30 @@ impl Handle {
                                          self.parser.clone_ref(), id)
     }
 
+    /// Get pointer to the method identified by its definition ID.
+    ///
+    /// Note that there might exist multiple definition IDs for the same method pointer, as
+    /// definition IDs include information about definition syntax whereas method pointer identifies
+    /// the desugared entity.
+    pub fn method_pointer
+    (&self, id:&dr::graph::Id) -> FallibleResult<language_server::MethodPointer> {
+        let defined_on_type = match id.crumbs.as_slice() {
+            [crumb] => {
+                if crumb.extended_target.is_empty() {
+                    self.path.module_name().to_string()
+                } else {
+                    crumb.extended_target.iter().map(|segment| segment.as_str()).join(".")
```

Comment:
is `as_str()` needed here? would `join` not work without it? If not, can we implement something to make it work?